cmd/kube-apiserver/app/server.go

```diff
@@ -586,17 +585,6 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 
 	// Set up encryption at rest via transformer overrides if a configuration file was provided.
 	if len(s.Etcd.EncryptionProviderConfigFilepath) != 0 {
-		cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
-		if err != nil {
-			return nil, err
-		}
-		// Returns an envelope Service implemented by a KMS supported by the cloud hosting k8s.
-		cloudKMSGetter := func(name string) (envelope.Service, error) {
-			return cloud.KeyManagementService(name)
-		}
-		// Register cloud in plugin registry in case a cloudprovided KMS provider is required.
-		encryptionconfig.KMSPluginRegistry.RegisterCloudProvidedKMSPlugin(cloudKMSGetter)
```

Comment:
moving the registration into the init makes sense to me, but we need to ensure the following:
* `cloudprovider.InitCloudProvider` is always called in the kubeapiserver when a cloud provider is specifed (currently it is only called to determine the external host and to set up ssh tunneling, both of which can be skipped), and is called early (similar to how we set up the controller context with the cloud provider in the controller manager prior to initializing any controller configs)